src/lib/Base/Container/ApiLoader/RepositoryConfigurationProvider.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\Base\Container\ApiLoader;
+
+use Ibexa\Bundle\Core\ApiLoader\Exception\InvalidRepositoryException;
+use Ibexa\Contracts\Core\Container\ApiLoader\RepositoryConfigurationProviderInterface;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+
+/**
+ * @phpstan-import-type TRepositoryConfiguration from RepositoryConfigurationProviderInterface
+ * @phpstan-import-type TRepositoryStorageConfiguration from RepositoryConfigurationProviderInterface
+ * @phpstan-import-type TRepositorySearchConfiguration from RepositoryConfigurationProviderInterface
+ *
+ * @phpstan-type TRepositoryListItemConfiguration array{
+ *      storage: TRepositoryStorageConfiguration,
+ *      search: TRepositorySearchConfiguration,
+ *      fields_groups: array{default: string, list: string[]},
+ *      options: array<string, mixed>
+ * }
+ * @phpstan-type TRepositoryListConfiguration array<string, TRepositoryListItemConfiguration>
+ */
+final class RepositoryConfigurationProvider implements RepositoryConfigurationProviderInterface
+{
+    private const string REPOSITORY_STORAGE = 'storage';
+    private const string REPOSITORY_CONNECTION = 'connection';
+    private const string DEFAULT_CONNECTION_NAME = 'default';
+
+    private ConfigResolverInterface $configResolver;
+
+    /** @phpstan-var TRepositoryListConfiguration */
+    private array $repositories;
+
+    /**
+     * @phpstan-param TRepositoryListConfiguration $repositories
+     */
+    public function __construct(ConfigResolverInterface $configResolver, array $repositories)
+    {
+        $this->configResolver = $configResolver;
+        $this->repositories = $repositories;
+    }
```

Comment:
```suggestion
final readonly class RepositoryConfigurationProvider implements RepositoryConfigurationProviderInterface
{
    private const string REPOSITORY_STORAGE = 'storage';
    private const string REPOSITORY_CONNECTION = 'connection';
    private const string DEFAULT_CONNECTION_NAME = 'default';

    /**
     * @phpstan-param TRepositoryListConfiguration $repositories
     */
    public function __construct(
        private ConfigResolverInterface $configResolver,
        private array $repositories,
    ) {
```

The same can be applied to other classes declared (or changed) in this PR.